ZipArchive.cs

```diff
@@ -248,20 +376,115 @@ public ZipEntry AddFile (string sourcePath, string archivePath = null, EntryPerm
 			return ReadEntry ((ulong)index);
 		}
 
+		/// <summary>
+		/// Adds the stream to the archive.
+		/// </summary>
+		/// <param name="entryName">The name of the entry with in the archive</param>
+		/// <param name="data">A stream containing the data to add to the archive</param>
+		/// <param name="method">The compression method to use</param>
+		public void AddEntry (string entryName, Stream data,
+			CompressionMethod method = CompressionMethod.DEFAULT)
+		{
+			AddStream (data, entryName, method: method);
+		}
+
+		/// <summary>
+		/// Adds the byte[] data to the archive
+		/// </summary>
+		/// <param name="entryName">The name of the entry with in the archive</param>
+		/// <param name="data">The draw byte[] array to add to the entry</param>
+		/// <param name="method">The compression method to use</param>
+		public void AddEntry (string entryName, byte [] data,
+			CompressionMethod method = CompressionMethod.DEFAULT)
+		{
+			AddEntry (data, entryName, method: method);
+		}
+
+		/// <summary>
+		/// Adds the text provided as a new entry within the archive. This is useful 
+		/// for when you want to just add plain text to the archive. For example a
+		/// README/LICENCE file which needs to be pre-processed rather than just adding
+		/// the raw file.
+		/// </summary>
+		/// <param name="entryName">The name of the entry with in the archive</param>
+		/// <param name="text">The text to add to the entry</param>
+		/// <param name="encoding">The Encoding to use for the data.</param>
+		/// <param name="method">The compression method to use</param>
+		public void AddEntry (string entryName, string text, Encoding encoding,
+			CompressionMethod method = CompressionMethod.DEFAULT)
+		{
+			AddEntry (entryName, encoding.GetBytes (text), method);
```

Comment:
Add null checks. `encoding` must not be null (or, better, should default to `Encoding.Default`) and passing null string to `GetBytes` will throw an error